api/debug.go

```diff
@@ -70,6 +72,26 @@ func (d *Debug) Profile(seconds int) ([]byte, error) {
 	return body, nil
 }
 
+// PProf returns a pprof profile for the specified number of seconds. The caller
+// is responsible for closing the returned io.ReadCloser once all bytes are read.
+func (d *Debug) PProf(ctx context.Context, name string, seconds int) (io.ReadCloser, error) {
+	r := d.c.newRequest("GET", "/debug/pprof/"+name)
+	r.ctx = ctx
+
+	// Capture a profile for the specified number of seconds
+	r.params.Set("seconds", strconv.Itoa(seconds))
+
+	_, resp, err := d.c.doRequest(r)
+	if err != nil {
+		return nil, fmt.Errorf("error making request: %s", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return nil, generateUnexpectedResponseCodeError(resp)
+	}
+	return resp.Body, nil
```

Comment:
What's the use case for returning the `resp.Body` raw vs `ioutil.ReadAll`-ing and returning `[]byte`?

Reply:
Good question! That is indeed a change from the previous methods.

The nice thing about returning an `io.ReadCloser` is that we don't have to buffer the entire response in memory. We can write the response to a file as a stream, so the memory of the CLI process should stay relatively constant.

With `ioutil.ReadAll` we have to read the entire profile into the CLI process memory, which could be quite large. I don't know if that's actually a probably in this case, but in general using an `io.Reader` or `io.ReadCloser` will give the caller the option to either stream or buffer, where as returning a `[]byte` forces them to use the fully buffered response.